dns.c

```diff
@@ -2,6 +2,9 @@
 #include <netdb.h>
 #include <sys/types.h>
 #include <netinet/in.h>
+#ifndef BIND_8_COMPAT
+#define BIND_8_COMPAT /* if Bind 9, Bind 8 compatibility */
```

Comment:
Simple, self-contained, can't think of any downsides. Obvious in retrospect, which is the hallmark of a hard-earned good choice.

Reply:
Don't forget a comment here to point fingers on MacOS.

Reply:
I've added a "Mac OS X:" prefix to the comment, I hope that satisfies.